src/app/clusters/general-commissioning-server/general-commissioning-server.cpp

```diff
@@ -239,34 +296,64 @@ bool emberAfGeneralCommissioningClusterCommissioningCompleteCallback(
         }
         else
         {
-            if (failSafe.NocCommandHasBeenInvoked())
+            CHIP_ERROR err;
+
+#if defined CHIP_CONFIG_TC_REQUIRED_ACKNOWLEDGEMENTS && defined CHIP_CONFIG_TC_REQUIRED_ACKNOWLEDGEMENTS_VERSION
+
+            uint16_t termsAndConditionsAcceptedAcknowledgements;
+            bool hasTermsAndConditionsRequiredAcknowledgementsBeenAccepted;
+            bool hasTermsAndConditionsRequiredAcknowledgementsVersionBeenAccepted;
```

Comment:
These could be simplified in name. These names have crossed into the "so many words it's hard to read". Imagine `hasRequiredTermAccepted` --> it's close enough.

Reply:
Updating